core/src/system_monitor_service.rs

```diff
@@ -254,52 +277,69 @@ pub fn verify_net_stats_access() -> Result<(), String> {
 
 #[cfg(target_os = "linux")]
 fn read_disk_stats() -> Result<DiskStats, String> {
-    let file_path_diskstats = PROC_DISKSTATS_PATH;
-    let file_diskstats = File::open(file_path_diskstats).map_err(|e| e.to_string())?;
-    let mut reader_diskstats = BufReader::new(file_diskstats);
-    parse_disk_stats(&mut reader_diskstats)
-}
-
-#[cfg_attr(not(target_os = "linux"), allow(dead_code))]
-fn parse_disk_stats(reader_diskstats: &mut impl BufRead) -> Result<DiskStats, String> {
     let mut stats = DiskStats::default();
     let mut num_disks = 0;
-    for line in reader_diskstats.lines() {
-        let line = line.map_err(|e| e.to_string())?;
-        let values: Vec<_> = line.split_ascii_whitespace().collect();
-
-        if values.len() != 20 {
-            return Err("parse error, expected exactly 20 disk stat elements".to_string());
-        }
-        if values[2].starts_with("loop") || values[1].ne("0") {
-            // Filter out the loopback io devices.
-            // Only look at raw device (filter partitions)
+    for blk_device_dir in std::fs::read_dir(SYS_BLOCK_PATH).map_err(|e| e.to_string())? {
+        let blk_device_dir = blk_device_dir.map_err(|e| e.to_string())?;
+        if blk_device_dir
+            .file_name()
+            .to_string_lossy()
+            .starts_with("loop")
```

Comment:
at least the dmcrypt volumes still exist in sysfs as `dm-[0-9]+`.  will check my other machine for the mdraid volumes in a bit, but almost positive they are there, both are virtual block devices.

this may work better as a filter chain on the iterator

Reply:
mdraid as `md[0-9]+`